### DISCUSSION TITLE
lower-dimensional meshes and VectorVariableComponentAux

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#17988] [Q&A General] [answered]
Hi everyone,

I populate a CONSTANT MONOMIAL_VEC AuxVariable from an AuxKernel successfully.  However, zeroes are erroneously returned when i use a VectorVariableComponentAux to extract components of it when I have meshes with mixed-dimensional elements:

- for 3D elements, everything is correct;
- for 2D elements, the "z" component of the VectorVariableComponentAux is always zero, irrespective of the z value of the MONOMIAL_VEC, even if `Mesh dimension` and `Spatial dimension` are 3;
- for 1D elements, the "y" and "z" components are zero.

I'm not sure whether this is a bug, or deliberate, or whether I have to provide a certain flag or input-file command to retrieve the correct value.

a

**Comment by GiudGiud** [comment]
@rwcarlsen will know.
I m curious too

**Comment by rwcarlsen** [comment, marked as answer]
Upon first glance, this seems like a bug.  I'd need to dig more into the code to see if there are any design issues that prevent us from providing a more sensible result.

**Reply by GiudGiud** [reply]
Created an issue so we dont loose track of it. This could be consequential to some people
